clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/agent/KubernetesCachingAgent.java

```diff
@@ -72,31 +79,87 @@
 
   @Getter protected final Long agentInterval;
 
+  protected final KubernetesConfigurationProperties configurationProperties;
+
+  protected final KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap;
+
   protected KubernetesCachingAgent(
       KubernetesNamedAccountCredentials namedAccountCredentials,
       ObjectMapper objectMapper,
       Registry registry,
       int agentIndex,
       int agentCount,
-      Long agentInterval) {
+      Long agentInterval,
+      KubernetesConfigurationProperties configurationProperties,
+      KubernetesSpinnakerKindMap kubernetesSpinnakerKindMap) {
     this.accountName = namedAccountCredentials.getName();
     this.credentials = namedAccountCredentials.getCredentials();
     this.objectMapper = objectMapper;
     this.registry = registry;
     this.agentIndex = agentIndex;
     this.agentCount = agentCount;
     this.agentInterval = agentInterval;
+    this.configurationProperties = configurationProperties;
+    this.kubernetesSpinnakerKindMap = kubernetesSpinnakerKindMap;
   }
 
   protected Map<String, Object> defaultIntrospectionDetails() {
     Map<String, Object> result = new HashMap<>();
     result.put("namespaces", getNamespaces());
-    result.put("kinds", primaryKinds());
+    result.put("kinds", kindsToCache());
     return result;
   }
 
   protected abstract List<KubernetesKind> primaryKinds();
 
+  protected abstract boolean cachesKind(KubernetesKind kind);
+
+  protected List<KubernetesKind> kindsToCache() {
+    List<KubernetesKind> cacheKinds;
+
+    if (configurationProperties.getCache().isCacheAll()) {
+      cacheKinds = primaryKinds();
+
+    } else if (configurationProperties.getCache().getCacheKinds() != null
+        && configurationProperties.getCache().getCacheKinds().size() > 0) {
+      // If provider config specifies what kinds to cache, use it
+      cacheKinds =
+          configurationProperties.getCache().getCacheKinds().stream()
+              .map(KubernetesKind::fromString)
+              .filter(this::cachesKind)
+              .filter(credentials::isValidKind)
+              .collect(Collectors.toList());
+
+    } else {
+      // Only cache kinds used in Spinnaker's classic infrastructure screens, which are the kinds
+      // mapped to Spinnaker kinds like ServerGroups, Instances, etc.
+      cacheKinds =
+          kubernetesSpinnakerKindMap.allKubernetesKinds().stream()
```

Comment:
Is it important to start with `kubernetesSpinnakerKindMap.allKubernetesKinds()` as opposed to `primaryKinds()` here?

Reply:
Or maybe even more directly, start with SPINNAKER_UI_KINDS, and call `translateSpinnakerKind` ... I guess we still need
```
              .filter(this::cachesKind)
              .filter(credentials::isValidKind)
```
but it seems simpler.